### PR TITLE
YarpFixManager: fix yarp fixture for compatibility with recent changes in libYARP_manager

### PR DESCRIPTION
**Pull request by Nicogene**
This PR introduces changes in `YarpFixManager` to make it compatible with recent changes in `libYARP_manager`, fixes robotology/gazebo-yarp-plugins#299 (TESTED) 

MERGE AFTER robotology/yarp#1244

### CODE REVIEW

File: src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp [hunk 1]
```diff
@@ -61,7 +61,7 @@ bool YarpFixManager::setup(int argc, char** argv) {
         enableRestrictedMode();
 
         // load the fixture (application xml)
-        char szAppName[255];
+        char* szAppName = new char [10];
```
Comment (traversaro) [inline src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp]: Why we moved from `255` to `10` ?

Comment (Nicogene) [inline src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp]: `255` was a random number, you can't know how much long the name is.
It is only a placeholder in fact the real allocation is made in `KnowledgeBase::addApplication` where we can know exactly how much long is the name.


Probably befor

Comment (drdanz) [inline src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp]: I just added the same comment in the RTF patch :smiley:

File: src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp [hunk 2]
```diff
@@ -72,6 +72,11 @@ bool YarpFixManager::setup(int argc, char** argv) {
                             "yarpmanager (loadApplication) cannot setup the fixture because " +
                             std::string(getLogger()->getFormatedErrorString()));
         initialized = true;
+        if (szAppName)
+        {
+            delete szAppName;
```
Comment (drdanz) [inline src/rtf-plugins/fixture-managers/yarpmanager/YarpFixManager.cpp]: `delete []`

**Pull request merged**